enhancements/multi-arch/heterogenous-architecture.md

```diff
@@ -0,0 +1,270 @@
+
+No changes are needed in the MCO for Phase 1 as the machine-os-content image would be a manifest listed image and the machine config daemon would extract the relevant architecture's machine-os-content based on the node it runs on. In the future architecture
+specific configurations should not be required, instead they need to be templatized and be generalized for certain items, for example kubelet system reserved memory could scale based on page size. 
+
+### Hypershift
+
+The addition of heterogenous support in Openshift also opens up a new dimension for [support][OCPPLAN-5684] in Hypershift where the control plane and the workers could run on different architectures. There would be certain changes needed to realize this as there are limitations
+which need to be addressed:
+
+- When creating the hosted cluster, it is needed to specify the `--release-image` option to specify the manifest listed release image in order to be able to deploy on a worker node of any architecture (as the default is an x86 release image). Even when that is the case
+there are some components like machine-approver, konnectivity-agent, etc..which have hardcoded x86 images.
+- Hypershift config would have to provide a flag to indicate that it is being installed on a heterogenous arch cluster in which case it will pull down images from the heterogenous release page rather than the default amd64 for the hosted cluster.
+- The component images of the hosted control plane which default to x86 will have to provide overrides at config time. Today these can be overridden through annotations, but it would be good to have it as part of the command.
+- Another option would be to mandate an x86 worker node be present for Hypershift and Hypershift would select the x86 node for hosted cluster deployment
+- When creating the nodepool, the defaults for instances types and AMIs are x86 specific. The instance types can be overridden but not the AMIs. An option needs to be added to the create cluster command to override AMIs
+
+### Console
+
+As of today console filters content based on control plane architecture. This needs to be entirely removed or made configurable for heterogenous clusters
+
+### Install
+
+For Phase 1, the way to add more workers to the cluster is to handcraft a MachineSet manifest file for a worker node and specify the architecture specific details like instance type, image references and deploy the machineset to get a worker node. This can be pseudo automated
+further by creating the manifests using `openshift-install create manifests` and modifying or adding worker manifest files. Eventually the installer should support being able to declare differing compute and control plane sections in the install-config mainly allowing
+architectures to be different which is not allowed today.
+
+### Open Questions
+
+Definition of GA - Plan is to GA phase 1 while documenting the limitations. This is easier than creating support exceptions or limiting payload access.
```

Comment:
this doesn't look like a question

Reply:
yup.removed and added some actual questions